### PR TITLE
board: stm32_min_dev: Add SPI support

### PR DESCRIPTION
**Pull request by jukkar**
This commit adds SPI support to stm32_dev_mini a.k.a Blue Pill
stm32f103 board.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

**Comment by jukkar** [comment]
Hi, I have a enc28j60 SPI based ethernet module and I am trying to connect it to stm32f103 board. Currently I am not able to send data and I am suspecting that this SPI code is not correct. Does this PR look anything sane or usable?

**Comment by ydamigos** [comment]
@jukkar Are you using pins PA4 or PB12 for CS? If yes, did you try to set NSS to:
```
STM32F1_PINMUX_FUNC_PA4_SPI1_MASTER_NSS_OE
STM32F1_PINMUX_FUNC_PB12_SPI2_MASTER_NSS_OE
```

File: boards/arm/stm32_min_dev/Kconfig.defconfig [hunk 1]
```diff
@@ -17,4 +17,14 @@ config UART_STM32_PORT_1
 
 endif # UART_CONSOLE
 
+if SPI
+
```
Comment (ydamigos) [inline boards/arm/stm32_min_dev/Kconfig.defconfig]: Did you try to enable interrupt mode?
```
config SPI_STM32_INTERRUPT
	default y
```

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

**Comment by jukkar** [comment]
> Are you using pins PA4 or PB12 for CS? If yes, did you try to set NSS to:

I have tried both for CS, and also now tried STM32F1_PINMUX_FUNC_PA4_SPI1_MASTER_NSS_OE. No luck with SPI in my setup.

File: boards/arm/stm32_min_dev/Kconfig.defconfig (hunk 1 above)
Comment (jukkar) [reply boards/arm/stm32_min_dev/Kconfig.defconfig]: Yes, have tried this too but that did not help.

**Comment by jukkar** [comment]
Of course I might have wrong settings for the enc28j60 chip, currently I have these configured in .conf file.

```
CONFIG_ETH_ENC28J60=y
CONFIG_ETH_ENC28J60_0=y
CONFIG_ETH_ENC28J60_0_SPI_BUS_FREQ=36000000
CONFIG_ETH_ENC28J60_0_SPI_PORT_NAME="SPI_1"
CONFIG_ETH_ENC28J60_0_GPIO_PORT_NAME="GPIOA"
CONFIG_ETH_ENC28J60_0_GPIO_PIN=15

```

I have tried different pins for GPIO but so far it had not made any difference.

**Comment by ydamigos** [comment]
> I have tried both for CS, and also now tried STM32F1_PINMUX_FUNC_PA4_SPI1_MASTER_NSS_OE. 

I have used SPI on olimexino_stm32 board (stm32f103rb) to access a SDHC card. The board doesn't use PA4 or PB12 as CS. It uses PD2 as CS and the SDHC over SPI driver controls the CS. Did you try to set CS using:
```
CONFIG_ETH_ENC28J60_0_GPIO_SPI_CS
CONFIG_ETH_ENC28J60_0_SPI_CS_PORT_NAME
CONFIG_ETH_ENC28J60_0_SPI_CS_PIN
```

**Comment by jukkar** [comment]
> I have used SPI on olimexino_stm32 board (stm32f103rb) to access a SDHC card. The board doesn't use PA4 or PB12 as CS. It uses PD2 as CS and the SDHC over SPI driver controls the CS. Did you try to set CS using:

Not yet, I will try that next. Thanks for the hint.

**Comment by jukkar** [comment]
Rebased and fixed the commit message. So far I have not been able to verify if these settings are correct and working.

**Comment by galak** [comment]
@jukkar what do we want to do with this PR?

**Comment by nashif** [comment]
@jukkar, ping

**Comment by jukkar** [comment]
Lets close this for time being, I cannot verify if this really works atm.

**Pull request closed without merging**